microsoft-integration/media-gallery-tool/src/main/java/org/sakaiproject/microsoft/mediagallery/controller/MainController.java

```diff
@@ -0,0 +1,483 @@
+		}
+		
+		MicrosoftAccessToken mcAccessToken = microsoftAuthorizationService.getAccessToken(userId);
+		model.addAttribute("delegatedClientConfigured", (mcAccessToken != null));
+		
+		if(mcAccessToken != null) {
+			model.addAttribute("mcUserAccount", mcAccessToken.getMicrosoftUserId());
+			
+			////first time, loading process could require some time, just show a loading spinner and items will be loaded by AJAX
+			if(mediaGallerySessionBean.isFirstTime()) {
+				mediaGallerySessionBean.setFirstTime(false);
+				return INDEX_WS_TEMPLATE;
+			}
+			
+			//when refreshing, call a redirect to clean all GET parameters from URL. Avoid "hard" refresh if user press F5
+			if(StringUtils.isNotBlank(refreshSection)) {
+				mediaGallerySessionBean.resetItems(MediaGallerySessionBean.Type.valueOf(refreshSection));
+				microsoftCommonService.resetDriveItemsCache();
+				microsoftCommonService.resetUserDriveItemsCache(userId);
+
+				redirectAttributes.addFlashAttribute("refreshSection", refreshSection);
+				redirectAttributes.addFlashAttribute("sortBy", sortBy);
+				redirectAttributes.addFlashAttribute("treeView", treeView);
+				mediaGallerySessionBean.setFirstTime(true);
+				
+				return REDIRECT_INDEX;
+			}
+			
+			MediaGallerySessionBean.Type reset = (MediaGallerySessionBean.Type)mediaGallerySessionBean.getReset();
+			Map<String, MicrosoftDriveItem> itemsMap = mediaGallerySessionBean.getItemsMap();
+			if(itemsMap.isEmpty() || reset != null) {
+				//USER items
+				if(reset == null || reset == MediaGallerySessionBean.Type.USER) {
+					mediaGallerySessionBean.addType(MediaGallerySessionBean.Type.USER, rb.getString(MediaGallerySessionBean.Type.USER.name()));
+
+					List<MicrosoftDriveItem> userItems = microsoftCommonService.getAllMyDriveItems(userId, filter);
+					if(userItems != null && !userItems.isEmpty()) {
+						mediaGallerySessionBean.getItemsByType().put(MediaGallerySessionBean.Type.USER, userItems);
+						
+						//iterate all items and add them to the session maps
+						exploreAndDoSomething(userItems, (i) -> mediaGallerySessionBean.addItem(MediaGallerySessionBean.Type.USER, i));
+					}
+				}
+				
+				//SHARED items
+				if(reset == null || reset == MediaGallerySessionBean.Type.SHARED) {
+					mediaGallerySessionBean.addType(MediaGallerySessionBean.Type.SHARED, rb.getString(MediaGallerySessionBean.Type.SHARED.name()));
+
+					List<MicrosoftDriveItem> sharedItems = microsoftCommonService.getAllMySharedDriveItems(userId, filter);
+					if(sharedItems != null && !sharedItems.isEmpty()) {
+						mediaGallerySessionBean.getItemsByType().put(MediaGallerySessionBean.Type.SHARED, sharedItems);
+						
+						//iterate all items and add them to the session maps
+						exploreAndDoSomething(sharedItems, (i) -> mediaGallerySessionBean.addItem(MediaGallerySessionBean.Type.SHARED, i));
+					}
+				}
+			}
+			
+			//sort items by name or date
+			Comparator<MicrosoftDriveItem> comparator = (a, b) -> {
```

Comment:
This code is the same on line 229. Refactor?

Reply:
Done